src/credentials/credentialsStore.ts

```diff
@@ -44,7 +44,7 @@ export class CredentialsStore {
 
         if (!credentials) {
             credentials = await this.setCredentials(credentialsProviderId, credentialsProvider)
-        } else if (credentialsProvider.getHashCode() !== credentials.credentialsHashCode) {
+        } else if (credentialsProvider.getHashCode() !== credentials.credentialsHashCode || credentialsProvider.isSsoProfile()) {
```

Comment:
Do we actually need to check `getHashCode` here ? What is the cost of _always_ doing `this.invalidateCredentials(credentialsProviderId)` when `upsertCredentials()` is called?

(To avoid log noise we could just skip the `getLogger().verbose()` call.)

Reply:
The hash code check is used to filter out profiles that have not changed. eg: If a file contains 5 profiles, but only one of the profiles was modified in that file edit, this filters out the other 4 profiles.